pkg/macvlan/macvlan.go

```diff
@@ -342,8 +343,40 @@ func macvlanCmdAdd(args *skel.CmdArgs) error {
 		return err
 	}
 
-	result.DNS = n.DNS
+	// Fetch existing default route on eth0 interface
+	existingLink, err := netlink.LinkByName("eth0")
+	if err != nil {
+		return fmt.Errorf("couldn't get interface eth0: %v", err)
+	}
+
+	routes, err := netlink.RouteList(existingLink, netlink.FAMILY_V4)
+
+	if err != nil {
+	       return err
+	}
+
+	// Delete existing default route
+	for _, route := range routes {
+		if route.Dst == nil{
+			netlink.RouteDel(&route)
```

Comment:
you shouldn't have to find the existing route; you can just construct a `Route` object with the right `LinkIndex` and `Dst` and then delete it

Reply:
Done!